src/atdgen_codec_runtime.ml

```diff
@@ -173,7 +173,36 @@ struct
     else
       Some (decode json)
 
-  let fieldOptional s f = optional (field s f)
+  exception MissingFieldError of string
```

Comment:
do we really need new exception? can make `field` return `option`?

Reply:
Changed, see https://github.com/ahrefs/bs-atdgen-codec-runtime/pull/11/commits/1fb4750c124f7e459776bc6214de2bf952ecf55d#diff-af319977d60814aacb356044a4666058R176